tiledb/sm/serialization/array_schema.cc

```diff
@@ -726,6 +775,16 @@ Status array_schema_to_capnp(
   timestamp_builder.set(0, timestamp_range.first);
   timestamp_builder.set(1, timestamp_range.second);
 
+  // Dimension labels
+  auto num_labels = array_schema.dim_label_num();
+  auto dim_labels_buidler =
```

Comment:
We should only call `initDimensionLabels` if `num_labels` > 0. This will prevent the `hasDimensionLabels` from triggering because an empty DimensionLabels object was initialized.